### ISSUE TITLE
[Bug]: Simple regex strings with variable replacement fail in MongoDB queries

### ISSUE THREAD

**Issue by sirbeagle** [appsmithorg/appsmith#17240]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When attempting to use a regex value for finding records against a MongoDB, a JavaScript expression gets misinterpreted and causes no records to be found.

In our case we are search an array of strings using the `$in` clause. In the simplest example when using `Find Document(s)`, we will see records returned with this:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/1271367/193292651-74156058-4aaa-4772-ae09-9cc924f33f97.png">

However, replacing `chip` with a simple JavaScript expression causes no records to be found:

<img width="640" alt="image" src="https://user-images.githubusercontent.com/1271367/193293651-a74bd953-30fa-45a2-a3aa-ea8f15e8c10b.png">

Including the slashes in the expressions also yields no results:

<img width="644" alt="image" src="https://user-images.githubusercontent.com/1271367/193293058-ee2143a4-b868-4b87-9eb3-3f1e77f7122b.png">

In each case, the evaluated value is the same. We are using `Find Document(s)` here but this has also been seen using `Aggregate`. No errors seem to be reported.

### Steps To Reproduce

1. Create a MongoDB collection where record(s) have a value that is an array of strings. In our case we have: 
```
{
  _id: ObjectId('SOMEVALUE'),
  additionalTitles: [
    "I am a string",
    "I am a string with the word chip in it.",
    "I am another chip string."
  ]
}
```
2. Create a `Find Document(s)` query like in the Description in an Appsmith app.

### Version

Self hosted - v1.8.3

**Comment by Nikhil-Nandagopal** [comment]
@sirbeagle this should work if you turn off smart substitution in the settings of the query

**Comment by sirbeagle** [comment]
@Nikhil-Nandagopal Yes, I can confirm - turning off the Smart BSON Substitution in the query settings allows a regex query to work:

```
{ additionalTitles: {$in: [ {{ '/chip/'}} ] }}
```

**Comment by nzidol** [comment]
Hmm regression??? I had to go through all my aggregations and turn the "smart" BSON switch off to get them to work again.... All worked around Sep 20th, usually pick up new Appsmith version within 3 or 4 days.

**Comment by Nikhil-Nandagopal** [comment]
@manish535 @sondermanish @sribalajig would the data type project not have solved this?

**Comment by sribalajig** [comment]
@subrata71 can you help explain the scope of datatype handling epic?

**Comment by sondermanish** [comment]
@Nikhil-Nandagopal The logic that handles this flow is associated with smart substitution which is separate from what datatype handling project worked with.
As a current workaround we can use this format: `{ <field>: { $regex: 'pattern', $options: '<options>' } }`  and bind the pattern without using the `/`.

**Comment by Nikhil-Nandagopal** [comment]
@sribalajig @sondermanish but this seems like a regression so what changed suddenly? it seems to have worked before Sep 20th

**Comment by Naveen-Goud** [comment]
hii @rohan-arthur @Nikhil-Nandagopal   @sirbeagle  , I am trying to reproduce this issue , will update the solution approrach after debugging .

**Comment by Naveen-Goud** [comment]
Hii @Nikhil-Nandagopal @rohan-arthur @sirbeagle , Picking up this issue !!

**Comment by rohan-arthur** [comment]
sure @Naveen-Goud ! Please tag @appsmithorg/query-js-pod for any communication
